src/main/java/de/tum/in/www1/artemis/web/rest/CompetencyResource.java

```diff
@@ -277,12 +326,22 @@ public ResponseEntity<List<CompetencyWithTailRelationDTO>> importAllCompetencies
         if (courseId == sourceCourseId) {
             throw new BadRequestAlertException("Cannot import from a course into itself", "Course", "courseCycle");
         }
-        var course = courseRepository.findByIdElseThrow(courseId);
+        var targetCourse = courseRepository.findByIdElseThrow(courseId);
         var sourceCourse = courseRepository.findByIdElseThrow(sourceCourseId);
-        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, course, null);
+        authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.INSTRUCTOR, targetCourse, null);
         authorizationCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.EDITOR, sourceCourse, null);
 
-        List<CompetencyWithTailRelationDTO> importedCompetencies = competencyService.importAllCompetenciesFromCourse(course, sourceCourse, importRelations);
+        var competencies = competencyRepository.findAllForCourse(sourceCourse.getId());
+        List<CompetencyWithTailRelationDTO> importedCompetencies;
+
+        if (importRelations) {
+            var relations = competencyRelationRepository.findAllWithHeadAndTailByCourseId(sourceCourse.getId());
+            importedCompetencies = competencyService.importCompetenciesAndRelations(targetCourse, competencies, relations);
+        }
+        else {
+            importedCompetencies = competencyService.importCompetencies(targetCourse, competencies).stream()
```

Comment:
Also, this else-branch is identical to the one above, maybe consider extracting this into a method?